sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -343,6 +344,87 @@ abstract class ExplodeBase extends UnaryExpression with CollectionGenerator with
   }
 }
 
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Separates the elements of array `expr` into multiple rows recursively.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(10, 20));
+       10
+       20
+      > SELECT _FUNC_(a) FROM VALUES (array(1,2)), (array(3,4)) AS v1(a);
+       1
+       2
+       3
+       4
+      > SELECT _FUNC_(a) FROM VALUES (array(array(1,2), array(3,4))) AS v1(a);
+       1
+       2
+       3
+       4
+  """,
+  since = "3.0.0")
+case class UnNest(child: Expression) extends UnaryExpression with Generator with CodegenFallback {
```

Comment:
Can we implement codegen as well?

Reply:
seems very difficult to do codegen

Reply:
Hm, why? similar expressions look being able to do it.

Reply:
for each round of unnesting, we have to know what dataType it is at that time.